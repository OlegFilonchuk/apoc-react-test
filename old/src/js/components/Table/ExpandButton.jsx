import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default function ExpandButton({ onExpand, isExpanded, type }) {
  const iconClassName = classNames({
    'sc-icon-right-arrow': !isExpanded,
    'sc-icon-down-arrow': isExpanded
  });

  return (
    <button onClick={onExpand} className="expand-btn" type={type}>
      <span className={iconClassName} />
    </button>
  );
}

ExpandButton.propTypes = {
  /**
   * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type
   */
  type: PropTypes.string,

  /**
   * It will be called if expand button is clicked
   */
  onExpand: PropTypes.func,

  /**
   * Show if row was expanded
   */
  isExpanded: PropTypes.bool.isRequired
};

ExpandButton.defaultProps = {
  type: 'button',
  className: '',
  onExpand: () => null
};
