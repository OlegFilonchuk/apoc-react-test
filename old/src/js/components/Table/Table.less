@import '~chameleon/less/variables.less';
@import './tableVariables.less';

.table-component {
  td,
  th {
    overflow-wrap: break-word;
  }

  th {
    &.sc-left-align {
      text-align: left;
    }

    &.sc-right-align {
      text-align: right;
    }

    &.sc-center-align {
      text-align: center;
    }
  }

  .empty-table-container td {
    background-color: @emptyTableContainerBackground;
    text-align: center;
    padding: 5px;
  }
}

.table-component-with-striped-rows {
  &__row {
    &--odd > td {
      background-color: @table-bg-accent;
    }
    &--even > td {
      background-color: @white-color;
    }
  }
}
