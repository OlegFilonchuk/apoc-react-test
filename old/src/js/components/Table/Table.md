## Necessary sub components are
* [TableBodyWithExpandRows](#tablebodywithexpandrows)
* [TableCellWithTextEllipsis](#tablecellwithtextellipsis)
* [TableCell](#tablecell)
* [TableHeadRow](#tableheadrow)
* [TableHeadCell](#tableheadcell)
* [TableRowWithExpandContent](#tablerowwithexpandcontent)
* [TableWithPagination](#tablewithpagination)

## Example of table
It uses [TableHeadRow](#tableheadrow)

    (
      <Table>
        <TableHeadRow>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRow>
        <tbody>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
        </tbody>
      </Table>
    )

## Example of table with striped rows
It uses [TableHeadRow](#tableheadrow)

    (
      <Table isRowsStriped={true}>
        <TableHeadRow>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRow>
        <tbody>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
          <tr>
            <td>Content</td>
            <td>Content</td>
            <td>Content</td>
          </tr>
        </tbody>
      </Table>
    )

## Example of table with 'emptyTableMessage'
It uses [TableHeadRow](#tableheadrow)

    (
      <Table emptyTableMessage="There are no any data">
        <TableHeadRow>
          <th>Title</th>
          <th>Title</th>
          <th>Title</th>
        </TableHeadRow>
      </Table>
    )
