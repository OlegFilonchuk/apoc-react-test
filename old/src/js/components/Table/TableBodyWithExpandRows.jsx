import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { isEqual } from 'lodash';

import './TableBodyWithExpandRows.less';

/* eslint-disable no-shadow */
export default class TableBodyWithExpandRows extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      expandedRowId: props.defaultExpandedRowId,
      expandedChildren: (props.isControlled && props.expandedChildren) || []
    };
  }

  componentWillReceiveProps(nextProps) {
    const child = React.Children.toArray(nextProps.children).find(child => child.props.isExpanded);

    if (child && child.props.id !== this.state.expandedRowId) {
      this.setState({ expandedRowId: child.props.id });
    }

    if (this.props.isControlled && !isEqual(nextProps.expandedChildren, this.props.expandedChildren)) {
      this.setState({ expandedChildren: nextProps.expandedChildren });
    }
  }

  /**
   * It should be child's prop id
   * @param {string} id
   */
  onExpand = id => {
    const { expandedRowId, expandedChildren } = this.state;
    const { allowMultipleExpanded } = this.props;

    if (!allowMultipleExpanded) {
      this.setState({ expandedRowId: expandedRowId === id ? null : id });
    }

    const child = React.Children.toArray(this.props.children).find(child => child.props.id === id);

    if (child.props.onExpand && !allowMultipleExpanded) {
      child.props.onExpand(child.props.id);
    } else if (allowMultipleExpanded && !expandedChildren.includes(child.props.id)) {
      this.setState({
        expandedChildren: [...expandedChildren, child.props.id]
      });
    } else if (allowMultipleExpanded && expandedChildren.includes(child.props.id)) {
      this.setState({
        expandedChildren: expandedChildren.filter(childId => childId !== child.props.id)
      });
    }
    if (this.props.isControlled) {
      this.props.onExpand(id);
    }
  };

  get tableBodyChildren() {
    return React.Children.map(this.props.children, child => {
      const cells = React.Children.toArray(child.props.children);
      const colSpan = this.props.customColspan ? this.props.customColspan : React.Children.count(cells);

      return [
        React.cloneElement(child, {
          className: this.getChildClassNames(child.props.id, child.props.className),
          onExpand: this.onExpand,
          isExpanded: this.isChildExpanded(child.props.id),
          expandOnClick: this.props.expandOnClick
        }),
        this.isChildExpanded(child.props.id) && (
          <tr className="expandable-table-row-content">
            <td colSpan={colSpan}>{child.props.expandContent}</td>
          </tr>
        )
      ];
    });
  }

  get classNames() {
    return classNames('table-body-with-expand-content', this.props.className);
  }

  getChildClassNames = (childId, childClassName) =>
    classNames('expandable-table-row-trigger', childClassName, {
      'expandable-table-row-trigger--expanded': this.isChildExpanded(childId)
    });

  isChildExpanded = childId =>
    !this.props.allowMultipleExpanded
      ? this.state.expandedRowId === childId
      : this.state.expandedChildren.includes(childId);

  render() {
    return this.props.customWrapper ? (
      this.props.customWrapper(props => (
        <tbody {...props} ref={props.innerRef} className={this.classNames}>
          {this.tableBodyChildren}
        </tbody>
      ))
    ) : (
      <tbody className={this.classNames}>{this.tableBodyChildren}</tbody>
    );
  }
}

TableBodyWithExpandRows.propTypes = {
  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * Array of <tr> elements
   */
  children: PropTypes.node,

  /**
   * TableRowWithExpandContent's id which should be expanded by default
   */
  defaultExpandedRowId: PropTypes.string,

  /**
   * If true a whole row will be clickable
   */
  expandOnClick: PropTypes.bool,

  /**
   * If true, rows don't be collapsed automatically
   */
  allowMultipleExpanded: PropTypes.bool,

  /**
   * Possibility to set custom number of colspan
   */
  customColspan: PropTypes.number,

  /**
   * Custom wrapper for table children. By default, table component and dependant expects particular children types
   * as they are operating on them. This props allow to add custom wrapper around them AFTER they are modified.
   */
  customWrapper: PropTypes.func,

  /**
   * Allow to controll collapse/expadned state from outside of component
   */
  isControlled: PropTypes.bool,
  /**
   * Function which allow to change outside collapse/expanded state. It gets id of clicked element.
   */
  onExpand: PropTypes.func,
  /**
   * String with id of element which is currently expanded (this should be passed only if isControlled is true and allowMultipleExpandedState is false)
   */
  expandedRowId: PropTypes.string,
  /**
   * Array of currently expanded elements ids in table (this should be passed only if isControlled and allowMultipleExpandedState are true)
   */
  expandedChildren: PropTypes.arrayOf(PropTypes.string)
};

TableBodyWithExpandRows.defaultProps = {
  defaultExpandedRowId: null,
  className: '',
  children: [],
  expandOnClick: false,
  allowMultipleExpanded: false,
  customColspan: null,
  customWrapper: null,
  isControlled: false,
  onExpand: () => {},
  expandedRowId: '',
  expandedChildren: []
};
