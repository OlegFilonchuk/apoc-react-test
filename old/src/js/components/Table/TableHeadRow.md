Purpose of this component is to be used together with [TableHeadCell](#tableheadcell). See example there.
