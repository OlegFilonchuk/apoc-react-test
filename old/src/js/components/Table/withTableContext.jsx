import React from 'react';
import propTypes from 'prop-types';

export const CONTEXT_ID = 'APOC_REACT_TABLE_CONTEXT';

export const tableContextPropTypes = propTypes.shape({
  /**
   * @type {function(uuid: string, index:number, contentTypes: CellContentTypes)}
   */
  registerColumn: propTypes.func,
  /**
   * @type {function(uuid: string)}
   */
  unregisterColumn: propTypes.func,
  /**
   * @type {function(index: ?number):CellContentTypes}
   */
  getColumnContentTypes: propTypes.func
});

/**
 * @type {CellContentTypes}
 */
export const defaultContentTypes = {
  dateContent: false,
  iconOrGraphicalContent: false,
  textContent: false,
  numericContent: false
};

export const defaultCellTableProps = {
  registerColumn: () => null,
  unregisterColumn: () => null,
  getColumnContentTypes: () => defaultContentTypes
};

export function withTableContext(Component) {
  function WithTableContext(props, context) {
    return <Component {...props} tableProps={context[CONTEXT_ID]} />;
  }

  WithTableContext.propTypes = Component.propTypes;

  WithTableContext.contextTypes = {
    [CONTEXT_ID]: tableContextPropTypes
  };

  WithTableContext.displayName = `withTableContext${Component.displayName || Component.name}`;

  WithTableContext.WrappedComponent = Component;

  return WithTableContext;
}

/* eslint-disable no-unused-vars */
/**
 * @typedef {{dateContent: boolean,iconOrGraphicalContent: boolean,textContent: boolean, numericContent: boolean}} CellContentTypes
 */
const CellContentTypes = {};
/* eslint-enable no-unused-vars */
