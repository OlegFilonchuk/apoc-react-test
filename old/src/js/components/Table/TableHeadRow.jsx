import React from 'react';
import PropTypes from 'prop-types';

import { markAsTableHeadRow } from './Table';

export default function TableHeadRow({ children, className }) {
  return (
    <thead className={className}>
      <tr>{children}</tr>
    </thead>
  );
}

markAsTableHeadRow(TableHeadRow);

TableHeadRow.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

TableHeadRow.defaultProps = {
  className: ''
};
