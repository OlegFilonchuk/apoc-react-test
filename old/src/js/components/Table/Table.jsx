import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import TableBodyWithExpandRows from './TableBodyWithExpandRows';
import { CONTEXT_ID, tableContextPropTypes, defaultContentTypes } from './withTableContext';

import './Table.less';

const tableHeadRowsList = [];

/**
 * @param Component
 */
export function markAsTableHeadRow(Component) {
  tableHeadRowsList.push(Component);
}

/**
 * @param element
 * @return {boolean}
 */
function isElementMarkedAsTableHead(element) {
  return tableHeadRowsList.includes(element.type);
}

/**
 * @param element
 * @return {boolean}
 */
function isElementMarkedAsTableBody(element) {
  return element.type === 'tbody' || element.type === TableBodyWithExpandRows;
}

export default class Table extends React.Component {
  // @todo: Never used, Should it be removed?
  static tableChildContextTypes = {
    [CONTEXT_ID]: tableContextPropTypes
  };
  // These two static member should stay like that until
  // https://github.com/reactjs/react-docgen/issues/317 is resolved
  static childContextTypes = Table.tableChildContextTypes;

  // @todo: Never used, Should it be removed?
  getChildContext = () => ({
    [CONTEXT_ID]: {
      registerColumn: (uuid, index, columnContentType) => {
        this.uuidToContentTypes.set(uuid, columnContentType);
        if (this.orderOfUuids[index]) {
          this.orderOfUuids.splice(index, 0, uuid);
        } else {
          this.orderOfUuids[index] = uuid;
        }
      },
      unregisterColumn: uuid => {
        const index = this.orderOfUuids.indexOf(uuid);

        this.uuidToContentTypes.delete(uuid);
        this.orderOfUuids.splice(index, 1);
      },
      getColumnContentTypes: index => {
        if (index == null) {
          return defaultContentTypes;
        }
        const uuid = this.orderOfUuids[index];
        const value = this.uuidToContentTypes.get(uuid);

        return value || defaultContentTypes;
      }
    }
  });

  // eslint-disable-next-line
  uuidToContentTypes = new Map();

  /** @type {{}[]} */
  orderOfUuids = [];

  get tableClassName() {
    return classNames('sc-table table-component', this.props.className, {
      'table-component-with-striped-rows': this.props.isRowsStriped
    });
  }

  get childrenArray() {
    return React.Children.toArray(this.props.children);
  }

  get tableHead() {
    return this.childrenArray.find(isElementMarkedAsTableHead);
  }

  get hasData() {
    return this.childrenArray.filter(isElementMarkedAsTableBody).some(tbody => Boolean(tbody.props.children));
  }

  getChildClassName = (row, index) =>
    classNames(row.props.className, {
      'table-component-with-striped-rows__row--odd': this.props.isRowsStriped && index % 2,
      'table-component-with-striped-rows__row--even': this.props.isRowsStriped && !(index % 2)
    });

  get colSpan() {
    return this.tableHead ? React.Children.count(this.tableHead.props.children) : 1;
  }

  getTableBodyChildren = children =>
    React.Children.map(children, (row, index) =>
      React.cloneElement(row, {
        className: this.getChildClassName(row, index)
      })
    );

  get tableBody() {
    return this.hasData ? (
      this.childrenArray.filter(isElementMarkedAsTableBody).map(tbody =>
        React.cloneElement(tbody, {
          children: this.getTableBodyChildren(tbody.props.children)
        })
      )
    ) : (
      <tfoot className="empty-table-container">
        <tr>
          <td colSpan={this.colSpan}>
            <div>{this.props.emptyTableMessage}</div>
          </td>
        </tr>
      </tfoot>
    );
  }

  render() {
    return (
      <table className={this.tableClassName}>
        {this.tableHead}
        {this.tableBody}
      </table>
    );
  }
}

Table.propTypes = {
  /**
   * If table should to show empty table message
   */
  isRowsStriped: PropTypes.bool,

  /**
   * List of custom classes added to component
   */
  className: PropTypes.string,

  /**
   * message which will be shown if table doesn't has content
   */
  emptyTableMessage: PropTypes.string,

  /**
   * table content
   */
  children: PropTypes.node.isRequired
};

Table.defaultProps = {
  isRowsStriped: false,
  emptyTableMessage: '',
  className: ''
};
