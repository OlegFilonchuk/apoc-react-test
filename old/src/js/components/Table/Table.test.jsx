import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Table from './Table';
import TableHeadRow from './TableHeadRow';
import { CONTEXT_ID } from './withTableContext';

const tableBody = (
  <tbody>
    <tr>
      <td>Content 1</td>
      <td>Content 2</td>
      <td>Content 3</td>
    </tr>
  </tbody>
);

const mountTable = ({ props = {}, mountMethod = mount, content = [] }) =>
  mountMethod(
    <Table {...props}>
      <TableHeadRow>
        <th>Title 1</th>
        <th>Title 2</th>
        <th>Title 3</th>
      </TableHeadRow>
      {content}
    </Table>
  );

describe('<Table />', () => {
  describe('should match snapshot', () => {
    it('with data', () => {
      expect(
        toJson(
          mountTable({
            content: tableBody,
            mountMethod: shallow
          })
        )
      ).toMatchSnapshot();
    });

    it('without data', () => {
      expect(toJson(mountTable({ mountMethod: shallow }))).toMatchSnapshot();
    });
  });

  describe('`emptyTableMessage` is ', () => {
    it("disable when table doesn't has any tr", () => {
      const table = mountTable({ content: tableBody });
      const emptyTableMessage = table.find('.empty-table-container');

      expect(emptyTableMessage.length).toBe(0);
    });

    it("available, `emptyTableMessage` shouldn't be rendered", () => {
      const table = mountTable({});
      const emptyTableMessage = table.find('.empty-table-container');

      expect(emptyTableMessage.length).toBe(1);
    });
  });

  describe('correctly set `className`', () => {
    let table;

    beforeEach(() => {
      table = mountTable({
        content: tableBody
      });
    });

    it('without passing `className`', () => {
      const element = table.find('.table-component');

      expect(element.length).toBe(1);
      expect(element).toHaveProp('className', 'sc-table table-component');
    });

    it('with passing `className`', () => {
      const element = table.find('.table-component');

      expect(element.length).toBe(1);
      table.setProps({ className: 'some-extra-class' });
      expect(element.getDOMNode().className).toContain('some-extra-class');
    });
  });

  describe('correctly set `emptyTableMessage`', () => {
    let table;

    beforeEach(() => {
      table = mountTable({
        props: { hasData: false }
      });
    });

    it('without passing `emptyTableMessage`', () => {
      const element = table.find('.empty-table-container div');

      expect(element).toHaveText('');
    });

    it('with passing `emptyTableMessage`', () => {
      const element = table.find('.empty-table-container div');
      const someText = 'Some text';

      table.setProps({ emptyTableMessage: someText });
      expect(element).toHaveText(someText);
    });
  });

  it('calculate colspan correctly', () => {
    const table = mountTable({
      props: { hasData: false }
    });
    const expectedColSpanValue = 3;
    const element = table.find('.empty-table-container td');

    expect(element.props().colSpan).toBe(expectedColSpanValue);
  });

  describe('with isRowStripped=true and three rows', () => {
    const createTableBodyWithThreeRows = key => (
      <tbody key={key}>
        <tr>
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
        </tr>
      </tbody>
    );

    describe('and three rows', () => {
      let table = null;

      beforeAll(() => {
        table = mountTable({
          props: { isRowsStriped: true },
          content: [createTableBodyWithThreeRows(1)]
        });
      });

      it('renders rows differently stripped', () => {
        const trs = table.find('tbody tr');

        expect(trs).toHaveLength(3);
        expect(trs.at(0)).toHaveClassName('table-component-with-striped-rows__row--even');
        expect(trs.at(1)).toHaveClassName('table-component-with-striped-rows__row--odd');
        expect(trs.at(2)).toHaveClassName('table-component-with-striped-rows__row--even');
      });
    });

    describe('and three rows in each of two tbody', () => {
      let table = null;

      beforeAll(() => {
        table = mountTable({
          props: { isRowsStriped: true },
          content: [createTableBodyWithThreeRows(1), createTableBodyWithThreeRows(2)]
        });
      });

      it('renders rows differently stripped, separately for each tbody', () => {
        const trs = table.find('tbody tr');

        expect(trs).toHaveLength(6);
        expect(trs.at(0)).toHaveClassName('table-component-with-striped-rows__row--even');
        expect(trs.at(1)).toHaveClassName('table-component-with-striped-rows__row--odd');
        expect(trs.at(2)).toHaveClassName('table-component-with-striped-rows__row--even');
        // for each tbody even/odd are calculated from scratch, that's why one even is after another in this case
        expect(trs.at(3)).toHaveClassName('table-component-with-striped-rows__row--even');
        expect(trs.at(4)).toHaveClassName('table-component-with-striped-rows__row--odd');
        expect(trs.at(5)).toHaveClassName('table-component-with-striped-rows__row--even');
      });
    });
  });

  describe('context', () => {
    describe('registerColumn ', () => {
      let table;
      let tableContext;

      beforeAll(() => {
        table = mountTable({
          content: tableBody,
          mountMethod: shallow
        });
        tableContext = table.instance().getChildContext()[CONTEXT_ID];
      });

      describe('starting with zero columns', () => {
        describe('few columns registered, one by one', () => {
          const columns = [
            { id: 'a', contentTypes: {} },
            { id: 'c', contentTypes: {} },
            { id: 'b', contentTypes: {} }
          ];

          beforeAll(() => {
            columns.forEach((x, k) => tableContext.registerColumn(x.id, k, x.contentTypes));
          });

          it('should save their content types in appropriate order', () => {
            expect(tableContext.getColumnContentTypes(0)).toBe(columns[0].contentTypes);
            expect(tableContext.getColumnContentTypes(1)).toBe(columns[1].contentTypes);
            expect(tableContext.getColumnContentTypes(2)).toBe(columns[2].contentTypes);
          });

          describe('few column registered later in the middle of array', () => {
            const nextColumns = [
              { id: 'f', contentTypes: {} },
              { id: 'h', contentTypes: {} }
            ];

            beforeAll(() => {
              tableContext.registerColumn(nextColumns[0].id, 1, nextColumns[0].contentTypes);
              tableContext.registerColumn(nextColumns[1].id, 3, nextColumns[1].contentTypes);
            });

            it('should save their content types in appropriate order', () => {
              expect(tableContext.getColumnContentTypes(0)).toBe(columns[0].contentTypes);
              expect(tableContext.getColumnContentTypes(1)).toBe(nextColumns[0].contentTypes);
              expect(tableContext.getColumnContentTypes(2)).toBe(columns[1].contentTypes);
              expect(tableContext.getColumnContentTypes(3)).toBe(nextColumns[1].contentTypes);
              expect(tableContext.getColumnContentTypes(4)).toBe(columns[2].contentTypes);
            });
          });
        });
      });
    });
  });
});
