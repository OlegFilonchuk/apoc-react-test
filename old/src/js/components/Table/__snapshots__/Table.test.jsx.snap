// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match snapshot with data 1`] = `
<table
  className="sc-table table-component"
>
  <TableHeadRow
    className=""
    key=".0"
  >
    <th>
      Title 1
    </th>
    <th>
      Title 2
    </th>
    <th>
      Title 3
    </th>
  </TableHeadRow>
  <tbody
    key=".1"
  >
    <tr
      className=""
      key=".0"
    >
      <td>
        Content 1
      </td>
      <td>
        Content 2
      </td>
      <td>
        Content 3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> should match snapshot without data 1`] = `
<table
  className="sc-table table-component"
>
  <TableHeadRow
    className=""
    key=".0"
  >
    <th>
      Title 1
    </th>
    <th>
      Title 2
    </th>
    <th>
      Title 3
    </th>
  </TableHeadRow>
  <tfoot
    className="empty-table-container"
  >
    <tr>
      <td
        colSpan={3}
      >
        <div />
      </td>
    </tr>
  </tfoot>
</table>
`;
